### PR TITLE
[Tests]snapshot.go: Remove use of RunVMAndExpectLaunchWithRunStrategy func

### PR DESCRIPTION
**Pull request by Barakmor1**
RunVMAndExpectLaunchWithRunStrategy should be deprecated as it is part of tests_utils.go and use expensive get and update API calls

Instead of Updating VM runStrategy in snapshot.go create vm with with the desired runStrategy.

### What this PR does
Before this PR:

After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by ShellyKa13 — COMMENTED** [review]
/lgtm

File: tests/storage/snapshot.go [hunk 1]
```diff
@@ -1293,49 +1293,59 @@ var _ = SIGDescribe("VirtualMachineSnapshot Tests", func() {
...
-				err = virtClient.CdiClient().CdiV1beta1().DataVolumes(vm.Namespace).Delete(context.Background(), volumeName, metav1.DeleteOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				Eventually(func() *metav1.Time {
-					pvc, err := virtClient.CoreV1().PersistentVolumeClaims(vm.Namespace).Get(context.Background(), volumeName, metav1.GetOptions{})
-					Expect(err).ToNot(HaveOccurred())
-					return pvc.DeletionTimestamp
-				}, 30*time.Second, time.Second).ShouldNot(BeNil())
+		It("vmsnapshot should update error if vmsnapshotcontent is unready to use and error", func() {
+			vm = renderVMWithRegistryImportDataVolume(cd.ContainerDiskAlpine, snapshotStorageClass)
+			runStrategyAlways := v1.RunStrategyAlways
```
Comment (dankenigsberg) [inline tests/storage/snapshot.go:1300]: I a future PR you can replace this with `libvmi.WithRunStrategy(v1.RunStrategyAlways)(vm)` or add `libvmi.Option...` to `renderVMWithRegistryImportDataVolume`

**Review by dankenigsberg — APPROVED** [review]

File: tests/storage/snapshot.go [hunk 2]
```diff
@@ -1293,49 +1293,59 @@ var _ = SIGDescribe("VirtualMachineSnapshot Tests", func() {
 
 				Expect(snapshot.Status.CreationTime).To(BeNil())
 			})
+		})
 
-			It("vmsnapshot should update error if vmsnapshotcontent is unready to use and error", func() {
-				vm = tests.RunVMAndExpectLaunchWithRunStrategy(virtClient, vm, v1.RunStrategyAlways)
```
Comment (enp0s3) [inline tests/storage/snapshot.go:1298]: @Barakmor1 Don't you need to wait for the VM to be ready? like in `RunVMAndExpectLaunchWithRunStrategy`
```
Eventually(ThisVM(updatedVM)).WithTimeout(300 * time.Second).WithPolling(time.Second).Should(BeReady())
```

Comment (Barakmor1) [reply tests/storage/snapshot.go:1298]: Yes, Thanks for the note will update the commit

**Review by enp0s3 — COMMENTED** [review]
@Barakmor1 The first commit in the PR seems not related to the PR scope. Can it be that its a leftover from some rebase?

**Comment by Barakmor1** [comment]
> @Barakmor1 The first commit in the PR seems not related to the PR scope. Can it be that its a leftover from some rebase?

Done :)

**Pull request merged**